tools/ssl/Makefile

```diff
@@ -1,41 +1,45 @@
-CA_SUB  = /C=PL/ST=Malopolska/L=Krakow/CN=MongooseIM Fake CA
-SRV_SUB = /C=PL/ST=Malopolska/L=Krakow/CN=MongooseIM Server
+all: mongooseim/server.pem mongooseim/cert.pem \
+  mongooseim/key.pem mongooseim/dh_server.pem \
+  mongooseim/pubkey.pem mongooseim/privkey.pem \
+  ca/cacert.pem ca/cakey.pem \
 
-all: ca/cacert.pem fake_cert.pem fake_pubkey.pem fake_privkey.pem fake_server.pem fake_dh_server.pem
+%/cacert.pem: openssl-%.cnf
+	mkdir -p $(@D)
+	openssl req -x509 -config $< -newkey rsa:2048 -sha256 -nodes -out $@ -outform PEM
 
-ca/cacert.pem ca/cakey.pem: openssl-ca.cnf
-	mkdir -p ca
-	-rm ca/cacert.pem
-	-rm ca/cakey.pem
-	openssl req -x509 -config $< -newkey rsa:2048 -sha256 -nodes \
-		-subj '$(CA_SUB)' -out ca/cacert.pem -outform PEM
+%/cakey.pem: %/cacert.pem;
 
-fake_cert.csr: openssl-server.cnf
-	openssl req -config $< -newkey rsa:2048 -sha256 -nodes -out $@ \
-		-subj '$(SRV_SUB)' -outform PEM
+%/cert.csr: openssl-%.cnf
+	mkdir -p $(@D)
+	openssl req -config $< -newkey rsa:2048 -sha256 -nodes \
+		-out $@ -keyout $(subst cert.csr,key.pem,$@) \
```

Comment:
Similarly to `%/cacert.pem: ...`, I think this rule should output `%/cert.csr %/key.pem` and use `$(@D)` automatic variable instead of `$(subst ...)`